packages/view/src/components/Statistics/AuthorBarChart/AuthorBarChart.tsx

```diff
@@ -0,0 +1,123 @@
+import type { ChangeEvent } from "react";
+import { useRef, useEffect, useState } from "react";
+import * as d3 from "d3";
+
+import type { ClusterNode } from "types";
+
+import type { AuthorDataType, MetricType } from "./AuthorBarChart.type";
+import { getDataByAuthor, sortDataByName } from "./AuthorBarChart.util";
+import { DIMENSIONS, METRIC_TYPE } from "./AuthorBarChart.const";
+
+import "./AuthorBarChart.scss";
+
+type AuthorBarChartProps = {
+  data: ClusterNode[];
+};
+
+const AuthorBarChart = ({ data: rawData }: AuthorBarChartProps) => {
+  const svgRef = useRef(null);
+  const [metric, setMetric] = useState<MetricType>(METRIC_TYPE[0]);
+
+  const authorData = getDataByAuthor(rawData);
+  const totalMetricValues = authorData.reduce(
+    (acc, item) => acc + item[metric],
+    0
+  );
+
+  useEffect(() => {
+    const data = authorData.sort((a, b) => {
+      if (a[metric] === b[metric]) {
+        return sortDataByName(a.name, b.name);
+      }
+      return a[metric] - b[metric];
+    });
+
```

Comment:
DIMENSIONS 객체의 프로퍼티에 접근하는 경우가 많은데, useEffect 화살표 콜백 함수 내부에서 스프레딩을 한 후에 가져다 쓰는 것은 어떨까용!!?

```suggestion
const { marigns, width, heights } = DIMENSIONS;
```

Reply:
DIMENSIONS와 같이 대문자 상수로 표현하면 상수임을 바로 알 수 있어 유용했는데 혹시 디스트럭처링으로 빼는 것이 더 좋으신가요? 
만약 그렇다면 내부 변수명 역시 `MARGINS` 로 전부 대문자 취급해야 하는 것이 어색하진 않은지 여쭤보고 싶습니다.

Reply:
개인적으로는 2번 이상 반복되는 경우 디스트럭쳐링해서 사용하는데, 중요한 내용은 아닌 것 같아서 추후 이야기나눈다음 리펙토링해도 좋을 듯 합니다 :)